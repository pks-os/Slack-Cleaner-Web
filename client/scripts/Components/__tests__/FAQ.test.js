import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import FAQ from '../FAQ';

const DEFAULT_PROPS = {
  onClose: jest.fn()
};

describe('<FAQ />', () => {
  it('Renders Correctly', () => {
    const renderer = createRenderer();
    const tree = renderer.render(<FAQ {...DEFAULT_PROPS} />);
    expect(tree).toMatchSnapshot();
  });

  it('does not render the button if no onClose function is passed', () => {
    const renderer = createRenderer();
    const tree = renderer.render(<FAQ />);
    expect(tree).toMatchSnapshot();
  });
});