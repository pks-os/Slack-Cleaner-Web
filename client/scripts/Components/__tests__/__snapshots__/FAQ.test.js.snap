// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FAQ /> Renders Correctly 1`] = `
<UNDEFINED>
  <Button
    classes="FAQ__Close"
    onClick={[MockFunction]}
    text="Close"
  />
  <h3>
    What is this thing?!
  </h3>
  <ul
    className="FAQ__List"
  >
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "How does this work?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "This app uses the <a href=\\"https://api.slack.com/methods\\" target=\\"_blank\\">Slack Web API Methods</a> that are open to all users. You define the type of files you want to search for, this app helps you navigate them, and the API handles the deletion.",
          }
        }
      />
    </li>
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Is this made by Slack?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Nope, it's made by <a href=\\"http://ngquad.com\\" target=\\"_blank\\">Drew Minns</a>, a developer from Toronto who would LOVE to work for Slack ;).",
          }
        }
      />
    </li>
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "In the other versions, there was a bulk delete. Where'd that go?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "The API method to delete a file requires one argument, the ID of the file to delete which means that the bulk deletion made multiple calls to the API at the same time. Slack has introduced <a href='https://api.slack.com/docs/rate-limits#tiers'>rate limits</a> on the number of times a method can be called per minute. While yes, it allows burts, if there was an error related to the rate limit, it's hard to communicate why to users. Rather than having errors be caused by people with loads of files, introducing a queuing system, or straight up taking the app down, I added in some features some people had been looking for.",
          }
        }
      />
    </li>
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Is this thing safe?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "I store nothing. The only information needed to log you in is stored is stored as a unique session. There are no databases, no caching (except on your own machine), anything.",
          }
        }
      />
    </li>
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "I have a feature/idea/request/job offer/how is this built?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "This entire product is open source. Feel free to poke around the repo and add some PR's <a href=\\"https://github.com/ngquad/slackcleaner\\">here</a>",
          }
        }
      />
    </li>
  </ul>
</UNDEFINED>
`;

exports[`<FAQ /> does not render the button if no onClose function is passed 1`] = `
<UNDEFINED>
  <h3>
    What is this thing?!
  </h3>
  <ul
    className="FAQ__List"
  >
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "How does this work?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "This app uses the <a href=\\"https://api.slack.com/methods\\" target=\\"_blank\\">Slack Web API Methods</a> that are open to all users. You define the type of files you want to search for, this app helps you navigate them, and the API handles the deletion.",
          }
        }
      />
    </li>
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Is this made by Slack?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Nope, it's made by <a href=\\"http://ngquad.com\\" target=\\"_blank\\">Drew Minns</a>, a developer from Toronto who would LOVE to work for Slack ;).",
          }
        }
      />
    </li>
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "In the other versions, there was a bulk delete. Where'd that go?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "The API method to delete a file requires one argument, the ID of the file to delete which means that the bulk deletion made multiple calls to the API at the same time. Slack has introduced <a href='https://api.slack.com/docs/rate-limits#tiers'>rate limits</a> on the number of times a method can be called per minute. While yes, it allows burts, if there was an error related to the rate limit, it's hard to communicate why to users. Rather than having errors be caused by people with loads of files, introducing a queuing system, or straight up taking the app down, I added in some features some people had been looking for.",
          }
        }
      />
    </li>
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "Is this thing safe?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "I store nothing. The only information needed to log you in is stored is stored as a unique session. There are no databases, no caching (except on your own machine), anything.",
          }
        }
      />
    </li>
    <li
      className="FAQ__List-Item"
    >
      <p
        className="FAQ__Question"
        dangerouslySetInnerHTML={
          Object {
            "__html": "I have a feature/idea/request/job offer/how is this built?",
          }
        }
      />
      <p
        className="FAQ__Answer"
        dangerouslySetInnerHTML={
          Object {
            "__html": "This entire product is open source. Feel free to poke around the repo and add some PR's <a href=\\"https://github.com/ngquad/slackcleaner\\">here</a>",
          }
        }
      />
    </li>
  </ul>
</UNDEFINED>
`;
